The example in the documentation doesn't seem to compile

I tried to compile the following program:
```rust
use glsl_lang::{ast, parse::Parse};

fn main() {
    // Some GLSL source to parse
    let source = r#"void main() {
        gl_FragColor = vec4(1., 0.5, 0.25, 1.);
    }"#;

    // Try parsing the source
    let ast = ast::TranslationUnit::parse(source);
    assert!(ast.is_ok());
}
```

in Cargo.toml:

```
glsl-lang = { version ="0.2.0", features = ["lexer-v2-min"] }
```

but the compilation fails (rustc 1.61 nightly):

```
error[E0283]: type annotations needed
  --> src\main.rs:10:15
   |
10 |     let ast = ast::TranslationUnit::parse(source);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for type parameter `L` declared on the associated function `parse`
   |
   = note: cannot satisfy `_: LangLexer<'_>`
note: required by a bound in `glsl_lang::parse::Parse::parse`
  --> C:\Users\Alexandre\.cargo\registry\src\github.com-1ecc6299db9ec823\glsl-lang-0.2.0\src\parse.rs:50:21
   |
50 |     fn parse<'i, L: LangLexer<'i>>(
   |                     ^^^^^^^^^^^^^ required by this bound in `glsl_lang::parse::Parse::parse`
help: consider specifying the type arguments in the function call
   |
10 |     let ast = ast::TranslationUnit::parse::<'i, L>(source);
   |                                          +++++++++
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -44,6 +44,9 @@ jobs:
             sed -i '/fuzz/d' Cargo.toml
           fi
 
+      - name: Run doc tests
+        run: cargo test --workspace -q --doc
+
       - name: Run lib tests
         run: cargo test --workspace -q --lib
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+## [0.2.1] - 2022-06-11
 ### Changed
 - Updated to Rust 2021 edition
 
@@ -60,7 +62,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [0.1.0] - 2021-04-05
 - Initial release
 
-[Unreleased]: https://github.com/vtavernier/glsl-lang/compare/v0.2.0...HEAD
+[Unreleased]: https://github.com/vtavernier/glsl-lang/compare/v0.2.1...HEAD
+[0.2.1]: https://github.com/vtavernier/glsl-lang/compare/v0.2.0...v0.2.1
 [0.2.0]: https://github.com/vtavernier/glsl-lang/compare/v0.1.3...v0.2.0
 [0.1.3]: https://github.com/vtavernier/glsl-lang/compare/v0.1.2...v0.1.3
 [0.1.2]: https://github.com/vtavernier/glsl-lang/compare/v0.1.1...v0.1.2

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- Updated to Rust 2021 edition
+
 ### Fixed
 - `glsl-lang`: Fix documentation example for v0.2.0
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Fixed
+- `glsl-lang`: Fix documentation example for v0.2.0
 
 ## [0.2.0] - 2022-02-07
 ### Added

--- a/fuzz/Cargo.toml
+++ b/fuzz/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 name = "glsl-lang-fuzz"
 version = "0.1.0"
-edition = "2018"
+edition = "2021"
 default-run = "pp-test"
 
 [[bin]]

--- a/fuzz/Cargo.toml
+++ b/fuzz/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "glsl-lang-fuzz"
-version = "0.1.0"
+version = "0.2.1"
 edition = "2021"
 default-run = "pp-test"
 

--- a/lang-cli/Cargo.toml
+++ b/lang-cli/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "glsl-lang-cli"
-version = "0.2.0"
+version = "0.2.1"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
 edition = "2021"
 license = "BSD-3-Clause"
@@ -13,8 +13,8 @@ keywords = ["glsl", "language", "parser", "ast", "cli"]
 categories = ["command-line-interface", "parser-implementations", "rendering"]
 
 [dependencies]
-glsl-lang = { version = "0.2.0", features = ["lexer-v2-full"] }
-lang-util = "0.2.0"
+glsl-lang = { version = "0.2.1", features = ["lexer-v2-full"] }
+lang-util = "0.2.1"
 argh = "0.1"
 
 serde_json = { version = "1.0", optional = true }

--- a/lang-cli/Cargo.toml
+++ b/lang-cli/Cargo.toml
@@ -2,7 +2,7 @@
 name = "glsl-lang-cli"
 version = "0.2.0"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
-edition = "2018"
+edition = "2021"
 license = "BSD-3-Clause"
 description = "CLI for the glsl-lang crate"
 homepage = "https://github.com/vtavernier/glsl-lang"

--- a/lang-lexer/Cargo.toml
+++ b/lang-lexer/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "glsl-lang-lexer"
-version = "0.2.0"
+version = "0.2.1"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
 edition = "2021"
 license = "BSD-3-Clause"
@@ -14,7 +14,7 @@ categories = ["parser-implementations", "rendering"]
 
 [dependencies]
 glsl-lang-types = "0.2.0"
-lang-util = { version = "0.2.0", features = ["lalrpop"] }
+lang-util = { version = "0.2.1", features = ["lalrpop"] }
 
 thiserror = "1.0"
 

--- a/lang-lexer/Cargo.toml
+++ b/lang-lexer/Cargo.toml
@@ -2,7 +2,7 @@
 name = "glsl-lang-lexer"
 version = "0.2.0"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
-edition = "2018"
+edition = "2021"
 license = "BSD-3-Clause"
 description = "GLSL language lexers"
 homepage = "https://github.com/vtavernier/glsl-lang"

--- a/lang-pp/Cargo.toml
+++ b/lang-pp/Cargo.toml
@@ -2,7 +2,7 @@
 name = "glsl-lang-pp"
 version = "0.2.0"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
-edition = "2018"
+edition = "2021"
 license = "BSD-3-Clause"
 description = "GLSL language preprocessor"
 homepage = "https://github.com/vtavernier/glsl-lang"

--- a/lang-pp/Cargo.toml
+++ b/lang-pp/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "glsl-lang-pp"
-version = "0.2.0"
+version = "0.2.1"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
 edition = "2021"
 license = "BSD-3-Clause"
@@ -14,7 +14,7 @@ categories = ["parser-implementations", "rendering"]
 build = "build.rs"
 
 [dependencies]
-lang-util = "0.2.0"
+lang-util = "0.2.1"
 
 string_cache = "0.8"
 thiserror = "1.0"

--- a/lang-pp/src/exts.rs
+++ b/lang-pp/src/exts.rs
@@ -1,4 +1,4 @@
-use std::{array::IntoIter, collections::HashMap};
+use std::collections::HashMap;
 
 use crate::types::type_names::TypeNameAtom;
 
@@ -51,7 +51,7 @@ impl Registry {
 impl Default for Registry {
     fn default() -> Self {
         Self {
-            extensions: IntoIter::new([
+            extensions: [
                 ExtensionSpec::new(ExtNameAtom::from("GL_3DL_array_objects"), vec![]),
                 ExtensionSpec::new(
                     ExtNameAtom::from("GL_AMD_gpu_shader_int16"),
@@ -566,7 +566,8 @@ impl Default for Registry {
                     ],
                 ),
                 ExtensionSpec::new(ExtNameAtom::from("GL_OVR_multiview"), vec![]),
-            ])
+            ]
+            .into_iter()
             .map(|spec| (spec.name.clone(), spec))
             .collect(),
         }

--- a/lang-pp/src/processor.rs
+++ b/lang-pp/src/processor.rs
@@ -1,4 +1,4 @@
-use std::{array::IntoIter, collections::HashMap};
+use std::collections::HashMap;
 
 use lang_util::{FileId, SmolStr};
 
@@ -232,10 +232,7 @@ impl<'r> ProcessorStateBuilder<'r> {
                 )
                 .chain(self.definitions.into_iter())
                 .map(|definition| Definition::Regular(definition.into(), FileId::default()))
-                .chain(
-                    IntoIter::new([Definition::Line, Definition::File, Definition::Version])
-                        .into_iter(),
-                )
+                .chain([Definition::Line, Definition::File, Definition::Version].into_iter())
                 .chain(self.registry.all().map(|spec| {
                     Definition::Regular(
                         Define::object(spec.name().as_ref().into(), one.clone(), true).into(),

--- a/lang-pp/src/processor/definition.rs
+++ b/lang-pp/src/processor/definition.rs
@@ -845,7 +845,7 @@ impl<'d> MacroInvocation<'d> {
             .into_iter()
             .group_by(Event::is_token)
             .into_iter()
-            .map(|(is_token, events)| {
+            .flat_map(|(is_token, events)| {
                 if is_token {
                     // A token sequence
                     // TODO: Prevent re-allocation
@@ -863,7 +863,6 @@ impl<'d> MacroInvocation<'d> {
                     events.collect()
                 }
             })
-            .flatten()
             .collect();
 
         subs_stack.remove(self.definition.name());

--- a/lang-quote/Cargo.toml
+++ b/lang-quote/Cargo.toml
@@ -2,7 +2,7 @@
 name = "glsl-lang-quote"
 version = "0.2.0"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
-edition = "2018"
+edition = "2021"
 license = "BSD-3-Clause"
 description = "Quoting proc-macro for glsl-lang"
 homepage = "https://github.com/vtavernier/glsl-lang"

--- a/lang-quote/Cargo.toml
+++ b/lang-quote/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "glsl-lang-quote"
-version = "0.2.0"
+version = "0.2.1"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
 edition = "2021"
 license = "BSD-3-Clause"
@@ -17,7 +17,7 @@ proc-macro = true
 path = "src/lib.rs"
 
 [dependencies]
-glsl-lang = { version = "0.2.0", default-features = false, features = ["lexer-v2-full"] }
+glsl-lang = { version = "0.2.1", default-features = false, features = ["lexer-v2-full"] }
 
 proc-macro2 = "1"
 quote = "1"

--- a/lang-types/Cargo.toml
+++ b/lang-types/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "glsl-lang-types"
-version = "0.2.0"
+version = "0.2.1"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
 edition = "2021"
 license = "BSD-3-Clause"
@@ -13,7 +13,7 @@ keywords = ["glsl", "language", "parser"]
 categories = ["rendering"]
 
 [dependencies]
-lang-util = "0.2.0"
+lang-util = "0.2.1"
 thiserror = "1.0"
 
 rserde = { version = "1.0", optional = true, features = ["derive"], package = "serde" }

--- a/lang-types/Cargo.toml
+++ b/lang-types/Cargo.toml
@@ -2,7 +2,7 @@
 name = "glsl-lang-types"
 version = "0.2.0"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
-edition = "2018"
+edition = "2021"
 license = "BSD-3-Clause"
 description = "GLSL language AST"
 homepage = "https://github.com/vtavernier/glsl-lang"

--- a/lang-util-derive/Cargo.toml
+++ b/lang-util-derive/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "lang-util-derive"
-version = "0.2.0"
+version = "0.2.1"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
 edition = "2021"
 license = "BSD-3-Clause"

--- a/lang-util-derive/Cargo.toml
+++ b/lang-util-derive/Cargo.toml
@@ -2,7 +2,7 @@
 name = "lang-util-derive"
 version = "0.2.0"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
-edition = "2018"
+edition = "2021"
 license = "BSD-3-Clause"
 description = "proc-macros for lang-util"
 homepage = "https://github.com/vtavernier/glsl-lang"

--- a/lang-util-derive/src/node_content/content_display.rs
+++ b/lang-util-derive/src/node_content/content_display.rs
@@ -110,7 +110,7 @@ pub(crate) fn node_content_display(
 
                     let quoted = quote_spanned! {
                         variant.span() =>
-                            Self::#name#fields=> {
+                            Self::#name #fields=> {
                                 write!(f, " `{}`", #vs)?;
                             }
                     };

--- a/lang-util-dev/Cargo.toml
+++ b/lang-util-dev/Cargo.toml
@@ -2,7 +2,7 @@
 name = "lang-util-dev"
 version = "0.2.0"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
-edition = "2018"
+edition = "2021"
 license = "BSD-3-Clause"
 description = "Development utilities for parsers implemented using lang-util"
 homepage = "https://github.com/vtavernier/glsl-lang"

--- a/lang-util-dev/Cargo.toml
+++ b/lang-util-dev/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "lang-util-dev"
-version = "0.2.0"
+version = "0.2.1"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
 edition = "2021"
 license = "BSD-3-Clause"

--- a/lang-util/Cargo.toml
+++ b/lang-util/Cargo.toml
@@ -2,7 +2,7 @@
 name = "lang-util"
 version = "0.2.0"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
-edition = "2018"
+edition = "2021"
 license = "BSD-3-Clause"
 description = "Utilities for implementing parsers and ASTs using lalrpop and Logos"
 homepage = "https://github.com/vtavernier/glsl-lang"

--- a/lang-util/Cargo.toml
+++ b/lang-util/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "lang-util"
-version = "0.2.0"
+version = "0.2.1"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
 edition = "2021"
 license = "BSD-3-Clause"
@@ -13,7 +13,7 @@ keywords = ["language", "parser", "ast"]
 categories = ["parsing"]
 
 [dependencies]
-lang-util-derive = "0.2.0"
+lang-util-derive = "0.2.1"
 line-span = "0.1"
 smol_str = "0.1"
 text-size = "1.1"

--- a/lang/Cargo.toml
+++ b/lang/Cargo.toml
@@ -2,7 +2,7 @@
 name = "glsl-lang"
 version = "0.2.0"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
-edition = "2018"
+edition = "2021"
 license = "BSD-3-Clause"
 description = "GLSL 4.6 language LALR parser and AST"
 homepage = "https://github.com/vtavernier/glsl-lang"

--- a/lang/Cargo.toml
+++ b/lang/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "glsl-lang"
-version = "0.2.0"
+version = "0.2.1"
 authors = ["Vincent Tavernier <vince.tavernier@gmail.com>"]
 edition = "2021"
 license = "BSD-3-Clause"
@@ -17,7 +17,7 @@ lalrpop-util = { version = "0.19.7", default-features = false, features = ["std"
 lazy_static = "1.4"
 thiserror = "1.0"
 
-lang-util = { version = "0.2.0", features = ["lalrpop"] }
+lang-util = { version = "0.2.1", features = ["lalrpop"] }
 glsl-lang-pp = { version = "0.2.0", optional = true }
 glsl-lang-lexer = { version = "0.2.0", features = ["lalrpop"] }
 glsl-lang-types = "0.2.0"

--- a/lang/README.md
+++ b/lang/README.md
@@ -15,7 +15,7 @@ elements.
 ### Parsing GLSL
 
 ```rust
-use glsl_lang::{ast, parse::Parse};
+use glsl_lang::{ast, parse::DefaultParse};
 
 // Some GLSL source to parse
 let source = r#"void main() {

--- a/lang/src/lib.rs
+++ b/lang/src/lib.rs
@@ -10,7 +10,7 @@
 //! ## Parsing GLSL
 //!
 //! ```
-//! use glsl_lang::{ast, parse::Parse};
+//! use glsl_lang::{ast, parse::DefaultParse};
 //!
 //! // Some GLSL source to parse
 //! let source = r#"void main() {

--- a/xtask/Cargo.toml
+++ b/xtask/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 name = "xtask"
 version = "0.1.0"
-edition = "2018"
+edition = "2021"
 
 [dependencies]
 anyhow = "1.0.42"

--- a/xtask/Cargo.toml
+++ b/xtask/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "xtask"
-version = "0.1.0"
+version = "0.2.1"
 edition = "2021"
 
 [dependencies]

